From bf9bef189dc898c0f98f31aba4aa15736f1f29bf Mon Sep 17 00:00:00 2001
From: Zygmunt Krynicki <me@zygoon.pl>
Date: Wed, 2 Aug 2017 17:43:47 +0200
Subject: [PATCH] cmd: mark arch as non-reexecing distro

Arch doesn't re-execute because snap-confine has incompatible snap mount
directory compiled in and we cannot use the binary from core. This can
be fixed by building a communication channel from snapd to snap-confine
but earlier attempts were stuck on design.

Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>
---
 cmd/cmd.go      | 2 +-
 cmd/cmd_test.go | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmd/cmd.go b/cmd/cmd.go
index bad50f6f0..ed7554d2a 100644
--- a/cmd/cmd.go
+++ b/cmd/cmd.go
@@ -67,7 +67,7 @@ func distroSupportsReExec() bool {
 		return false
 	}
 	switch release.ReleaseInfo.ID {
-	case "fedora", "centos", "rhel", "opensuse", "suse", "poky":
+	case "fedora", "centos", "rhel", "opensuse", "suse", "poky", "arch":
 		logger.Debugf("re-exec not supported on distro %q yet", release.ReleaseInfo.ID)
 		return false
 	}
diff --git a/cmd/cmd_test.go b/cmd/cmd_test.go
index e95969eba..7e8d3277a 100644
--- a/cmd/cmd_test.go
+++ b/cmd/cmd_test.go
@@ -145,7 +145,7 @@ func (s *cmdSuite) TestNonClassicDistroNoSupportsReExec(c *C) {
 	// no distro supports re-exec when not on classic :-)
 	for _, id := range []string{
 		"fedora", "centos", "rhel", "opensuse", "suse", "poky",
-		"debian", "ubuntu",
+		"debian", "ubuntu", "arch",
 	} {
 		restore = release.MockReleaseInfo(&release.OS{ID: id})
 		defer restore()
-- 
2.13.3

